### PR TITLE
620 - Update util std functions

### LINKED ISSUE #620: Command throws an error when connecting with pipe
### Description

When connecting the commands with pipe after the command, it will throws an error

`./bin/run transaction:sign '{"type":0,"timestamp":123,"senderPublicKey":"abcd1234","amount":"100","asset":{}}' | echo`

```
events.js:183
      throw er; // Unhandled 'error' event
      ^
Error: write EPIPE
    at WriteWrap.afterWrite [as oncomplete] (net.js:868:14)
```

**Reproduces how often:** [What percentage of the time does it reproduce?]
Always

### Versions

2.0-

### PR DESCRIPTION
**Pull request by mitsujutsu**
### What was the problem?

Related to PR: https://github.com/LiskHQ/lisk-commander/pull/622
Race condition in `getRawStdin ` causing inconsistent results when piping.

### How did I fix it?

Update getStdin func, remove getRawStdin and promise.race. 

### How to test it?

./bin/run transaction:create -t=0 100 100L --passphrase="pass:123" | ./bin/run transaction:sign --passphrase="pass:123"

### Review checklist

* The PR resolves #620
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: src/utils/input/utils.js [hunk 1]
```diff
@@ -44,12 +44,15 @@ export const splitSource = source => {
...
+const timeoutPromise = readFromStd =>
 	new Promise((resolve, reject) => {
 		const id = setTimeout(() => {
 			clearTimeout(id);
-			reject(new Error(`Timed out after ${ms} ms`));
-		}, ms);
+			if (stdinIsTTY()) {
+				reject(new Error(`Timed out after ${DEFAULT_TIMEOUT} ms`));
+			}
+			resolve(readFromStd);
```
Comment (shuse2) [inline src/utils/input/utils.js]: I think it's strange to `resolve` a Promise here.

File: src/utils/input/utils.js [hunk 2]
```diff
@@ -44,13 +44,17 @@ export const splitSource = source => {
...
-	new Promise((resolve, reject) => {
-		const id = setTimeout(() => {
-			clearTimeout(id);
-			reject(new Error(`Timed out after ${ms} ms`));
-		}, ms);
-	});
+const timeoutPromise = async readFromStd => {
+	try {
+		await (() =>
+			new Promise(([reject]) => setTimeout(reject, DEFAULT_TIMEOUT)))();
```
Comment (shuse2) [inline src/utils/input/utils.js]: Shouldn't we `clearTimeout` in normal scenario?
Also, this doesn't throw error, so what are we doing for `try-catch`...? do we need `Promise` here?

Comment (shuse2) [reply src/utils/input/utils.js]: I haven't thought about how to clean it up, but intended functionality is something like this?
Maybe we can chain this function to `handleClose` part so it uses `getRawStdIn` in `getStdIn` and it just handles the output nicely?
```
const readFromStd = new Promise((resolve, reject) => {
	const rl = readline.createInterface({ input: process.stdin });
	const lines = [];
	rl
		.on('line', line => lines.push(line))
		.on('close', () => resolve(lines));

	const id = setTimeout(() => {
		clearTimeout(id);
		if (stdinIsTTY()) {
			reject(new Error(`Timed out after ${DEFAULT_TIMEOUT} ms`));
		} else {
			resolve(lines);
		}
	}, DEFAULT_TIMEOUT);
});
return readFromStd;
```

File: src/utils/input/utils.js [hunk 3]
```diff
@@ -80,10 +61,18 @@ export const getStdIn = ({
...
 		const lines = [];
 		const rl = readline.createInterface({ input: process.stdin });
 
+		// Prevent readline hanging when command called with no input or piped
+		const id = setTimeout(() => {
+			clearTimeout(id);
+			if (stdinIsTTY()) {
+				return reject(new Error(`Timed out after ${DEFAULT_TIMEOUT} ms`));
+			}
+			return resolve(lines);
```
[the remark this answers is not in the document]
Comment (mitsujutsu) [reply src/utils/input/utils.js]: Would it be cleaner to have separate timeout function and Promise.race as before? I realize we would need to declare all the input variables twice in the current setup, once before the timeout check and once on handleClose event to update them.

[the remark this answers is not in the document]
Comment (shuse2) [reply src/utils/input/utils.js]: I think it's probably easier to change `handleClose` to function `(line: string[]) => { passphrase,  second... }`, then call from `close` event and `timeout` event

**Pull request merged**